tests/end_to_end_cases/persistence.rs

```diff
@@ -82,6 +82,98 @@ async fn test_chunk_are_removed_from_memory_when_soft_limit_is_hit() {
     );
 }
 
+#[tokio::test]
+async fn test_chunks_are_removed_only_until_soft_limit() {
+    // Ensure that when chunks are evicted, only enough are evicted to get under the limit
+    let fixture = ServerFixture::create_shared().await;
+    let mut management_client = fixture.management_client();
+    let mut write_client = fixture.write_client();
+
+    let db_name = rand_name();
+    create_quickly_persisting_database(&db_name, fixture.grpc_channel()).await;
+
+    // setup soft limit to 1M (so we can write in all chunks without hitting that)
+    let mut lifecycle_rules = LifecycleRules {
+        mutable_linger_seconds: 1,
+        mutable_size_threshold: 100,
+        buffer_size_soft: 1024 * 1024, // 1MB
+        buffer_size_hard: 1024 * 1024, // 1MB
+        persist: true,
+        ..Default::default()
+    };
+
+    let mut rules = management_client.get_database(&db_name).await.unwrap();
+    rules.lifecycle_rules = Some(lifecycle_rules.clone());
+    management_client
+        .update_database(rules.clone())
+        .await
+        .unwrap();
+
+    // write in 10 chunks, and expect that some are persisted
+    // (as of time of writing, 10 chunks took up ~800K)
+    let num_chunks: usize = 10;
+    for _ in 0..num_chunks {
+        let lp_lines: Vec<_> = (0..1_000)
+            .map(|i| format!("data,tag1=val{} x={} {}", i, i * 10, i))
+            .collect();
+        let num_lines_written = write_client
+            .write(&db_name, lp_lines.join("\n"))
+            .await
+            .expect("successful write");
+        assert_eq!(num_lines_written, 1000);
+    }
+
+    // Ensure ensure that 10 chunks have been persisted
```

Comment:
Really really sure? ;)

```suggestion
    // Ensure that 10 chunks have been persisted
```